samples/javascript_nodejs/52.teams-messaging-extensions-search-auth-config/bots/teamsMessagingExtensionsSearchAuthConfigBot.js

```diff
@@ -0,0 +1,184 @@
+                            value: process.env.SiteUrl + '/public/searchSettings.html?settings=' + escapedSettings
+                        }
+                    ]
+                }
+            }
+        };
+    }
+
+    async handleTeamsMessagingExtensionConfigurationSetting(context, settings) {
+        // When the user submits the settings page, this event is fired.
+        if (settings.state != null) {
+            await this.userConfigurationProperty.set(context, settings.state);
+
+            await this.userState.saveChanges(context, false);
+        }
+    }
+
+    async handleTeamsMessagingExtensionQuery(context, query) {
+        const searchQuery = query.parameters[0].value;
+        const attachments = [];
+
+        const userSettings = await this.userConfigurationProperty.get(context, '');
+        if (userSettings && userSettings.includes('email')) {
+            // When the Bot Service Auth flow completes, the query.State will contain a magic code used for verification.
+            let magicCode = '';
+            if (query.state && Number.isInteger(Number(query.state))) {
+                magicCode = query.state;
+            }
+
+            var tokenResponse = await context.adapter.getUserToken(context, this.connectionName, magicCode);
+            if (!tokenResponse || !tokenResponse.token) {
+                // There is no token, so the user has not signed in yet.
+
+                // Retrieve the OAuth Sign in Link to use in the MessagingExtensionResult Suggested Actions
+                const signInLink = await context.adapter.getSignInLink(context, this.connectionName);
+
+                return {
+                    composeExtension: {
+                        type: 'auth',
+                        suggestedActions: {
+                            actions: [{
+                                type: 'openUrl',
+                                value: signInLink,
+                                title: 'Bot Service OAuth'
+                            }]
+                        }
+                    }
+                };
+            }
+
+            // The user is signed in, so use the token to create a Graph Clilent and search their email
+            const graphClient = new SimpleGraphClient(tokenResponse.token);
+            var messages = await graphClient.searchMailInbox(searchQuery);
+
+            // Here we construct a ThumbnailCard for every attachment, and provide a HeroCard which will be
+            // displayed if the selects that item.
+            messages.value.forEach(msg => {
+                const heroCard = CardFactory.heroCard(msg.from.emailAddress.address);
+                heroCard.content.subtitle = msg.subject;
+                heroCard.content.text = msg.body.content;
+                const heroCardContent = heroCard.content;
+
+                const preview = CardFactory.thumbnailCard(msg.from.emailAddress.address,
+                    msg.subject + '<br />' + msg.bodyPreview,
```

Comment:
perhaps a template literal is nicer than adding strings

Reply:
Thanks @johnataylor ... fixed these and cleaned things up a little.